sdk/include/reactos/ghostwnd.h

```diff
@@ -0,0 +1,20 @@
+/*
+ * PROJECT:     ReactOS header
+ * LICENSE:     GPL-2.0-or-later (https://spdx.org/licenses/GPL-2.0-or-later)
+ * PURPOSE:     Ghost window
+ * COPYRIGHT:   Copyright 2018 Katayama Hirofumi MZ (katayama.hirofumi.mz@gmail.com)
+ */
+#ifndef REACTOS_GHOST_WND_INCLUDED
+#define REACTOS_GHOST_WND_INCLUDED
+
+#define GHOSTCLASSNAME L"Ghost"
+#define GHOST_PROP L"GhostProp"
+
+typedef struct GHOST_DATA
+{
+    HWND hwndTarget;
+    HBITMAP hbm32bpp;
+    BOOL bDestroyTarget;
+} GHOST_DATA;
```

Comment:
If this header is meant to be used **ONLY** by user32 and win32k, then the best place is inside `win32ss/include`.

Reply:
Another question that popped out from my mind:
Are you sure they use also that shared data on Windows (Vista+...) to implement the ghosted windows?
Because it looks to me that there is the possibility for an attacker to wreak havoc in win32k by just modifying the window properties of a ghost window from user-mode, since this data is set by user-mode (user32) but retrieved/used by kernel-mode (win32k).

I would like to hear from @ThFabba / @learn-more opinions about that.

Reply:
Ghost window will be designed as its window procedure is running in user32. Its data should be accessible from user32.